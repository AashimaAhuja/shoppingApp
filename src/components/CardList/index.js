import React from 'react';
import Card from '../Card';

import './style.css';

const CardList = ({products,onProductLike}) => {
    return (
        <div className='productList'>
            {products && products.map(product => <Card product={product} key={product.id} onProductLike={onProductLike}/>)}
        </div>
    )
}

export default CardList;