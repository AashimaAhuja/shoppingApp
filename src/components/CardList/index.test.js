import { shallow } from 'enzyme';
import * as React from 'react';
import CardList from './';

const defaultProps = {
    products: [
        {
            img: 'imageUrl',
            name: 'Prod 1',
            brand: 'Brand 1',
            price: 30,
            size: 30,
            sold: false,
            id: 1,
            isLiked: false
        },
        {
            img: 'imageUrl',
            name: 'Prod 1',
            brand: 'Brand 1',
            price: 30,
            size: 30,
            sold: false,
            id: 2,
            isLiked: false
        },
        {
            img: 'imageUrl',
            name: 'Prod 1',
            brand: 'Brand 1',
            price: 30,
            size: 30,
            sold: false,
            id: 3,
            isLiked: false
        },
        {
            img: 'imageUrl',
            name: 'Prod 1',
            brand: 'Brand 1',
            price: 30,
            size: 30,
            sold: false,
            id: 4,
            isLiked: false
        },
    ],
    onProductLike: jest.fn()
}

describe('<CardList/> Tests', () => {
    it('should render CardList Component', () => {
        const CardListComponent = shallow(<CardList />)
        expect(CardListComponent).toMatchSnapshot();
    });
    it('should render CardList Component with default Props', () => {
        const CardListComponent = shallow(<CardList {...defaultProps} />)
        expect(CardListComponent).toMatchSnapshot();
    });
})
