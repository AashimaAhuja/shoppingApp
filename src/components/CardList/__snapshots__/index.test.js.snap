// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CardList/> Tests should render CardList Component 1`] = `
<div
  className="productList"
/>
`;

exports[`<CardList/> Tests should render CardList Component with default Props 1`] = `
<div
  className="productList"
>
  <Card
    key="1"
    onProductLike={[MockFunction]}
    product={
      Object {
        "brand": "Brand 1",
        "id": 1,
        "img": "imageUrl",
        "isLiked": false,
        "name": "Prod 1",
        "price": 30,
        "size": 30,
        "sold": false,
      }
    }
  />
  <Card
    key="2"
    onProductLike={[MockFunction]}
    product={
      Object {
        "brand": "Brand 1",
        "id": 2,
        "img": "imageUrl",
        "isLiked": false,
        "name": "Prod 1",
        "price": 30,
        "size": 30,
        "sold": false,
      }
    }
  />
  <Card
    key="3"
    onProductLike={[MockFunction]}
    product={
      Object {
        "brand": "Brand 1",
        "id": 3,
        "img": "imageUrl",
        "isLiked": false,
        "name": "Prod 1",
        "price": 30,
        "size": 30,
        "sold": false,
      }
    }
  />
  <Card
    key="4"
    onProductLike={[MockFunction]}
    product={
      Object {
        "brand": "Brand 1",
        "id": 4,
        "img": "imageUrl",
        "isLiked": false,
        "name": "Prod 1",
        "price": 30,
        "size": 30,
        "sold": false,
      }
    }
  />
</div>
`;
